vnet/router.go

```diff
@@ -171,14 +180,25 @@ func (r *Router) Start() error {
 	go func() {
 	loop:
 		for {
-			err := r.onProcessChunks()
+			d, err := r.processChunks()
 			if err != nil {
-				r.log.Warnf("[%s] %s", r.name, err.Error())
+				r.log.Errorf("[%s] %s", r.name, err.Error())
+				break
 			}
-			select {
-			case <-r.pushCh:
-			case <-cancelCh:
-				break loop
+
+			if d <= 0 {
+				select {
+				case <-r.pushCh:
+				case <-cancelCh:
+					break loop
```

Comment:
What does the `break loop` do vs doing nothing at all?

Both times you will fall back into the `for` loop I believe

Reply:
It's one of the confusing syntaxes in Go - and that's how you break out the for-loop, unlike to label for go-to..

* https://play.golang.org/p/HYqlhyNExD3
* https://medium.com/golangspec/labels-in-go-4ffd81932339